common/scala/src/main/scala/whisk/core/connector/Message.scala

```diff
@@ -53,7 +54,8 @@ case class ActivationMessage(
     activationId: ActivationId,
     activationNamespace: EntityPath,
     content: Option[JsObject],
-    cause: Option[ActivationId] = None)
+    cause: Option[ActivationId] = None,
+    traceMetadata: Option[SpanMetadata] = None)
```

Comment:
@rabbah and @ddragosd  My apologies for delayed response guys, I was travelling so couldn't respond earlier.

Here is Kafka message JSON with SpanMetadata details (under traceMetadata key).

`{"activationId":"efe25cca34af41ec9de6c6c06e1166f5","revision":"1-61cb0057d6b1cddd315f1998da518ca3","activationNamespace":"guest","transid":[25,1496053731735],"content":{},"traceMetadata":{"traceId":5910003036728227668,"spanId":5752436967903800978,"forceSampling":true},"action":{"path":"guest","name":"hello","version":"0.0.1"},"user":{"subject":"guest","namespace":"guest","authkey":"23bc46b1-71f6-4ed5-8c54-816aa4f8c502:123zO3xZCLrMN6v2BKK1dXYFpXlPkccOFqm12CdAsMgRU4VrNZ9lyGVCGuMDGIwP","rights":["READ","PUT","DELETE","ACTIVATE"]}}`